pinot-controller/src/main/java/org/apache/pinot/controller/BaseControllerStarter.java

```diff
@@ -297,6 +297,18 @@ private void setupHelixClusterConstraints() {
             MAX_STATE_TRANSITIONS_PER_INSTANCE, constraintItem);
   }
 
+  /**
+   * Creates an instance of PinotHelixResourceManager.
+   * <p>
+   * This method can be overridden by subclasses to instantiate the object
+   * with subclasses of PinotHelixResourceManager.
+   * By default, it returns a new PinotHelixResourceManager using the current configuration.
+   *
+   * @return A new instance of PinotHelixResourceManager.
+   */
+  protected PinotHelixResourceManager createPinotHelixResourceManager() {
```

Comment:
(minor) To be consistent with getter
```suggestion
  protected PinotHelixResourceManager createHelixResourceManager() {
```